client/src/js/modules/visits/views/visit_list.vue

```diff
@@ -0,0 +1,216 @@
+    name:'Visits',
+    components: {
+    'custom-table-component': CustomTableComponent,
+    'custom-table-row':CustomTableRow,
+    'pagination-panel': Pagination,
+    },
+    props: {
+        'visit' : String,
+    },
+    data() {
+        return {
+            totalRecords:  10,
+            pageSize: 15,
+            currentPage: 1,
+            proposal: app.prop,
+            visitCollection: [],
+            visits: [],
+            searchVisit : '',
+            headers: [
+                {
+                    key: "STARTDATE",
+                    title: 'Start'
+                },
+                {
+                    key: "ENDDATE",
+                    title: 'End'
+                },
+                {
+                    key: "VIS",
+                    title: 'Number'
+                },
+                {
+                    key: "BEAMLINENAME",
+                    title: 'Beamline'
+                },
+                {
+                    key: "LC",
+                    title: 'Local Contact'
+                },
+                {
+                    key: "COMMENTS",
+                    title: 'Comments'
+                },
+                {
+                    key: "DCCOUNT",
+                    title: 'Data Collections'
+                },
+                {
+                    key: "SESSIONTYPE",
+                    title: 'Type'
+                },
+                {
+                    key: "ARCHIVED"
+                }
+
+            ]
+
+        }
+    },
+    created() {
+        this.fetchData()
+    },
+    methods: {
+        async fetchData() {
+            // fetches visit data based on prop
+            this.visitCollection = new VisitCollection()
+            this.visitCollection.queryParams = { page: this.currentPage, per_page: this.pageSize };
+
+            this.visitCollection.queryParams.prop = this.proposal;
+            this.visitCollection.queryParams.s = this.searchVisit;
+
+            const results = await this.$store.dispatch('getCollection', this.visitCollection);
+            this.visits = results.toJSON().map((e) => {
+                return { ...e, clicked: false , edited_comment: e.COMMENTS};
+            });
+
+            this.totalRecords = results.state.totalRecords;
+
+        },
+        handlePageChange(data) {
+            try {
+
+            this.currentPage = data.currentPage;
+            this.pageSize = data.pageSize;
+
+            this.fetchData();
+
+            }
+            catch(e) {
+                window.onerror('Cannot go to selected page: ' + e);
+
+                return ;
+            }
+
+        },
+        async onEnter(visit) {
+            // when changing comments for visits
+            visit.COMMENTS = visit.edited_comment;
+
+            visit.clicked = false;
+
+            await this.$store.dispatch('updateDataToApi', {
+            url: '/proposal/visits/' + visit.VISIT + '?prop=' + visit.PROPOSAL + '&COMMENTS=' + visit.COMMENTS,
```

Comment:
Maybe consider using [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) instead of string concatenation, as it would result in easier to read code. For example, this would become:

`/proposal/visits/${visit.VISIT}?prop=${visit.PROPOSAL}&COMMENTS=${visit.COMMENTS}`